skrifa/src/outline/glyf/hint/engine/graphics_state.rs

```diff
@@ -0,0 +1,1130 @@
+    /// and <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.c#L4209>
+    pub(super) fn op_gfv(&mut self) -> OpResult {
+        let vector = self.graphics_state.freedom_vector;
+        self.value_stack.push(vector.x)?;
+        self.value_stack.push(vector.y)
+    }
+
+    /// Set reference point 0.
+    ///
+    /// SRP0[] (0x10)
+    ///
+    /// Pops: p (point number)
+    ///
+    /// Pops a point number from the stack and sets rp0 to that point number.
+    ///
+    /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#set-reference-point-0>
+    /// and <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.c#L4224>
+    pub(super) fn op_srp0(&mut self) -> OpResult {
+        let p = self.value_stack.pop_usize()?;
+        self.graphics_state.rp0 = p;
+        Ok(())
+    }
+
+    /// Set reference point 1.
+    ///
+    /// SRP1[] (0x11)
+    ///
+    /// Pops: p (point number)
+    ///
+    /// Pops a point number from the stack and sets rp1 to that point number.
+    ///
+    /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#set-reference-point-1>
+    /// and <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.c#L4238>
+    pub(super) fn op_srp1(&mut self) -> OpResult {
+        let p = self.value_stack.pop_usize()?;
+        self.graphics_state.rp1 = p;
+        Ok(())
+    }
+
+    /// Set reference point 2.
+    ///
+    /// SRP2[] (0x12)
+    ///
+    /// Pops: p (point number)
+    ///
+    /// Pops a point number from the stack and sets rp2 to that point number.
+    ///
+    /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#set-reference-point-2>
+    /// and <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.c#L4252>
+    pub(super) fn op_srp2(&mut self) -> OpResult {
+        let p = self.value_stack.pop_usize()?;
+        self.graphics_state.rp2 = p;
+        Ok(())
+    }
+
+    /// Set zone pointer 0.
+    ///
+    /// SZP0[] (0x13)
+    ///
+    /// Pops: n (zone number)
+    ///
+    /// Pops a zone number, n, from the stack and sets zp0 to the zone with
+    /// that number. If n is 0, zp0 points to zone 0. If n is 1, zp0 points
+    /// to zone 1. Any other value for n is an error.
+    ///
+    /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#set-zone-pointer-0>
+    /// and <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.c#L4746>
+    pub(super) fn op_szp0(&mut self) -> OpResult {
+        let n = self.value_stack.pop()?;
+        self.graphics_state.zp0 = n.try_into()?;
+        Ok(())
```

Comment:
could just,

```suggestion
        self.graphics_state.zp0 = n.try_into()
```

I imagine?

Reply:
This ends up with a return type of `()` while the function expects `Result<(), _>` :(